tests/test_mozilla_reps.py

```diff
@@ -0,0 +1,250 @@
+        testObj.assertEqual(events[0]['data']['local_start'], '2012-06-06T10:00:00')
+        testObj.assertEqual(events[0]['data']['start'], '2012-06-06T02:00:00')
+        testObj.assertEqual(events[0]['origin'], MOZILLA_REPS_SERVER_URL)
+        testObj.assertEqual(events[0]['uuid'], 'e701d4ed3b954361383d678d2168a44307d7ff60')
+        testObj.assertEqual(events[0]['updated_on'], 1339326000.0)
+
+        if len(events) > 1:
+            testObj.assertEqual(events[1]['data']['estimated_attendance'], 50)
+            testObj.assertEqual(events[1]['data']['local_start'], '2012-06-09T09:00:00')
+            testObj.assertEqual(events[1]['data']['start'], '2012-06-09T07:00:00')
+            testObj.assertEqual(events[1]['origin'], MOZILLA_REPS_SERVER_URL)
+            testObj.assertEqual(events[1]['uuid'], 'e83e53b1f1039f62c66d2a977e23df3adf5bdc87')
+            testObj.assertEqual(events[1]['updated_on'], 1339344000.0)
+
+
+class TestMozillaRepsBackend(unittest.TestCase):
+    """MozillaReps backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        mozilla_reps = MozillaReps(MOZILLA_REPS_SERVER_URL, origin='test')
+
+        self.assertEqual(mozilla_reps.url, MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(mozilla_reps.origin, 'test')
+        self.assertIsInstance(mozilla_reps.client, MozillaRepsClient)
+
+        # When origin is empty or None it will be set to
+        # the value in url
+        mozilla_reps = MozillaReps(MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(mozilla_reps.url, MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(mozilla_reps.origin, MOZILLA_REPS_SERVER_URL)
+
+        mozilla_reps = MozillaReps(MOZILLA_REPS_SERVER_URL, origin='')
+        self.assertEqual(mozilla_reps.url, MOZILLA_REPS_SERVER_URL)
+        self.assertEqual(mozilla_reps.origin, MOZILLA_REPS_SERVER_URL)
+
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether the events are returned"""
+
+        HTTPServer.routes()
+
+        # Test fetch events with their reviews
+        mozilla_reps = MozillaReps(MOZILLA_REPS_SERVER_URL)
+
+        events = [page for page in mozilla_reps.fetch()]
+
+        self.assertEqual(len(events), 20)
+
+        HTTPServer.check_events_contents(self, events)
+
+
+    @httpretty.activate
+    def test_fetch_empty(self):
+        """Test whether it works when no jobs are fetched"""
+
+        HTTPServer.routes(empty=True)
+
+        mozilla_reps = MozillaReps(MOZILLA_REPS_SERVER_URL)
+        events = [event for event in mozilla_reps.fetch()]
+
+        self.assertEqual(len(events), 0)
+
+class TestMozillaRepsBackendCache(unittest.TestCase):
+    """MozillaReps backend tests using a cache"""
+
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
```

Comment:
You also have to check the number of requests send to the server to know whether it really retrieved the data from the cache or not.